File: benchmark/php/mean_variance.php
<?php
$arr = [];
for ($i = 0; $i < 1000000; $i++) { $arr[] = $i + 1; }
$n = 1000000;
$start = hrtime(true);
$sum = 0;
for ($i = 0; $i < $n; $i++) { $sum += $arr[$i]; }
$mean = $sum / $n;
$sq = 0;
for ($i = 0; $i < $n; $i++) { $d = $arr[$i] - $mean; $sq += $d * $d; }
$var = $sq / $n;
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Mean: " . $mean . "
";
echo "Variance: " . $var . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/perl/mean_variance.pl
use Time::HiRes qw(time);
my @arr = 1..1000000;
my $n = 1000000;
my $start = time();
my $sum = 0;
for my $i (0..$n-1) { $sum += $arr[$i]; }
my $mean = $sum / $n;
my $sq = 0;
for my $i (0..$n-1) { my $d = $arr[$i] - $mean; $sq += $d * $d; }
my $var = $sq / $n;
my $end = time();
my $ms = ($end - $start) * 1000;
print "Mean: $mean
";
print "Variance: $var
";
printf "Time: %.0fms
", $ms;
